File: src/main/java/com/cruftlab/words/controller/WordsController.java
package com.cruftlab.words.controller;

import com.cruftlab.words.model.Word;
import com.cruftlab.words.service.WordService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class WordsController {
    private final WordService wordService;

    @Autowired
    public WordsController(WordService wordService) {
        this.wordService = wordService;
    }

    @RequestMapping(value = "/words/all", method = RequestMethod.GET)
    public Iterable<Word> getAll() {
        return wordService.findAll();
    }
}

File: src/main/kotlin/com/cruftlab/words/controller/WordsController.kt
package com.cruftlab.words.controller

import com.cruftlab.words.model.Word
import com.cruftlab.words.repository.WordRepository
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController


@RestController
class WordsController(val wordRepository: WordRepository) {
    @GetMapping(value = "/api/words/all")
    fun getAll(): Iterable<Word> {
        return wordRepository.findAll()
    }

    @GetMapping(value = "/api/words/count")
    fun getCount(): Long {
        return wordRepository.count()
    }
}
